homeassistant/components/braviatv/media_player.py

```diff
@@ -116,56 +116,31 @@ class BraviaTVMediaPlayer(CoordinatorEntity, MediaPlayerEntity):
     def __init__(self, coordinator, name, unique_id, device_info):
         """Initialize the entity."""
 
-        self._name = name
-        self._unique_id = unique_id
-        self._device_info = device_info
+        self._attr_device_info = device_info
+        self._attr_is_volume_muted = coordinator.muted
+        self._attr_media_content_id = coordinator.channel_name
+        self._attr_media_duration = coordinator.duration
+        self._attr_name = name
+        self._attr_source = coordinator.source
+        self._attr_source_list = coordinator.source_list
+        self._attr_unique_id = unique_id
 
         super().__init__(coordinator)
 
-    @property
-    def name(self):
-        """Return the name of the device."""
-        return self._name
-
-    @property
-    def unique_id(self):
-        """Return a unique_id for this entity."""
-        return self._unique_id
-
-    @property
-    def device_info(self):
-        """Return the device info."""
-        return self._device_info
-
     @property
     def state(self):
         """Return the state of the device."""
         if self.coordinator.is_on:
             return STATE_PLAYING if self.coordinator.playing else STATE_PAUSED
         return STATE_OFF
 
-    @property
-    def source(self):
-        """Return the current input source."""
-        return self.coordinator.source
-
-    @property
-    def source_list(self):
-        """List of available input sources."""
-        return self.coordinator.source_list
-
     @property
     def volume_level(self):
         """Volume level of the media player (0..1)."""
         if self.coordinator.volume is not None:
             return self.coordinator.volume / 100
         return None
```

Comment:
I think we can move this to `_refresh_volume` in the coordinator and store already calculated volume.

Reply:
Done